haystack/retriever/tfidf.py

```diff
@@ -64,14 +65,12 @@ def _calc_scores(self, query):
         )
         return indices_and_scores
 
-    def retrieve(self, query, candidate_doc_ids=None, top_k=10, verbose=True):
+    def retrieve(self, query, filters=None, top_k=10, verbose=True):
         # get scores
```

Comment:
Shall we maybe raise an exception or warning in case filters are supplied here as they won't be used?